framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java

```diff
@@ -2985,11 +2999,14 @@ protected boolean messageKeyMatches(
      * Return true if the element has an {@code @AnnotatedFor} annotation, for this checker or an
      * upstream checker that called this one.
      *
+     * <p>If the element is null or the errors should not be suppressed, directly returns false.
+     *
      * @param elt the source code element to check, or null
      * @return true if the element is annotated for this checker or an upstream checker
      */
     private boolean isAnnotatedForThisCheckerOrUpstreamChecker(@Nullable Element elt) {
-        if (elt == null || !useConservativeDefault("source")) {
+        if (elt == null
+                || (!useConservativeDefault("source") && !issueErrorsForOnlyAnnotatedForScope)) {
```

Comment:
This method supposedly answers the question "isAnnotatedForThisCheckerOrUpstreamChecker". Why does it matter whether we use conservative defaults or if errors in an annotatedfor scope should be suppressed?